### PR TITLE
[android] Remove redundant exceptions

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Several packages were creating their own exceptions for handling a missing activity, all with different messages, but we provide one in core. Also, the way we access the currentActivity was not consistent

# How
Fixed the inconsistencies and removed redundant exceptions.
Add a new property on `AppContext` `throwingActivity`

# Test Plan
bare-expo. Affected packages behave as before

### CODE REVIEW

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt [hunk 1]
```diff
@@ -126,7 +126,7 @@ class ContactsModule : Module() {
     get() = appContext.permissions ?: throw Exceptions.PermissionsModuleNotFound()
 
   private val activity: Activity
-    get() = appContext.activityProvider?.currentActivity ?: throw Exceptions.MissingActivity()
+    get() = appContext.currentActivity ?: throw Exceptions.MissingActivity()
```
Comment (aleqsio) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:129]: could we maybe move that exception to the `appContext.currentActivity` getter?

Comment (alanjhughes) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:129]: Something like this?
```kotlin
override val currentActivity: Activity
  get() {
    val activity = activityProvider?.currentActivity
      ?: (reactContext as? ReactApplicationContext)?.currentActivity
    
    return activity ?: throw Exceptions.MissingActivity()
  }
```

Comment (aleqsio) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:129]: I like it, now we don't need that boilerplate :)

Comment (lukmccall) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:129]: I like it, but perhaps we need to introduce a different getter for that. We can't change the current one. In some places, we actually check if the activity is null or not.

Comment (alanjhughes) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:129]: Any suggestions for what we would should name a new property if we don't change this one? 
```kotlin
val activity: Activity
  get() {
    val current = activityProvider?.currentActivity
      ?: (reactContext as? ReactApplicationContext)?.currentActivity
    
    return current ?: throw Exceptions.MissingActivity()
  }
```

**Review by lukmccall — APPROVED** [review]

**Pull request merged**